core/src/main/java/jenkins/security/ResourceDomainRootAction.java

```diff
@@ -222,7 +224,8 @@ public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
 
             try (ACLContext ignored = ACL.as(auth)) {
                 try {
-                    Stapler.getCurrent().invoke(req, rsp, Jenkins.get(), requestUrlSuffix + restOfPath);
+                    String path = requestUrlSuffix + Arrays.stream(restOfPath.split("[/]")).map(Util::rawEncode).collect(Collectors.joining("/"));
```

Comment:
```suggestion
                    String path = requestUrlSuffix + Arrays.stream(restOfPath.split("/")).map(Util::rawEncode).collect(Collectors.joining("/"));
```

Reply:
I find my variant clearer for characters that have even a slight chance to be regex metacharacters.

Reply:
Technically `/` has a zero chance of being a regex metacharacter but I think I understand what you mean. :-)